hedera-node/hedera-app/src/test/java/com/hedera/node/app/tss/TssBaseServiceTest.java

```diff
@@ -17,16 +17,145 @@
 package com.hedera.node.app.tss;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
 import static org.mockito.Mockito.doCallRealMethod;
+import static org.mockito.Mockito.lenient;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
 
+import com.hedera.hapi.node.state.addressbook.Node;
+import com.hedera.hapi.node.state.roster.Roster;
+import com.hedera.hapi.node.state.roster.RosterEntry;
+import com.hedera.node.app.spi.AppContext;
+import com.hedera.node.app.spi.store.StoreFactory;
+import com.hedera.node.app.spi.workflows.HandleContext;
+import com.hedera.node.app.store.ReadableStoreFactory;
+import com.hedera.node.app.tss.api.TssLibrary;
+import com.swirlds.platform.state.service.WritableRosterStore;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TssBaseServiceTest {
+    @Mock
+    private HandleContext handleContext;
+
+    @Mock
+    private AppContext appContext;
+
+    @Mock
+    private StoreFactory storeFactory;
+
+    @Mock
+    private ReadableStoreFactory readableStoreFactory;
+
+    @Mock
+    private WritableRosterStore rosterStore;
+
+    @Mock
+    private TssLibrary tssLibrary;
+
+    private TssBaseService subject;
+
+    @BeforeEach
+    void setUp() {
+        given(appContext.gossip()).willReturn(mock(AppContext.Gossip.class));
+
+        subject = new TssBaseServiceImpl(
+                appContext, mock(ExecutorService.class), mock(Executor.class), tssLibrary, mock(Executor.class));
+    }
 
-class TssBaseServiceTest {
     @Test
     void nameIsAsExpected() {
         final var subject = mock(TssBaseService.class);
         doCallRealMethod().when(subject).getServiceName();
         assertEquals(TssBaseService.NAME, subject.getServiceName());
     }
+
+    @Test
+    @DisplayName("Service won't set the current candidate roster as the new candidate roster")
+    void doesntSetSameCandidateRoster() {
```

Comment:
Are these unit tests duplicated from the `NodeStakeUpdatesTest`? They look extremely similar; maybe I didn't put them in the right place. I'm not seeing any tests removed from that class in this PR though..

Reply:
Originally I had these tests here, but didn't want to mock out all the tss message behavior 😅

Reply:
I think they are still calling `setCandidateRoster` , so must be in correct place :)

Reply:
These tests are failing in CI for the same reason I saw locally––all the TSS message behavior afterwards isn't mocked out

```
Cannot invoke "com.swirlds.config.api.Configuration.getConfigData(java.lang.Class)" because the return value of "com.hedera.node.app.spi.workflows.HandleContext.configuration()" is null
java.lang.NullPointerException: Cannot invoke "com.swirlds.config.api.Configuration.getConfigData(java.lang.Class)" because the return value of "com.hedera.node.app.spi.workflows.HandleContext.configuration()" is null
```

I checked with `NodeStakeUpdatesTest`, the display names are exactly the same. We can add these tests here too, if you'd like, but it'll be more work to mock the TSS messages.

Reply:
FYI I pushed a small commit that eliminated the duplicate node and roster objects between the two tests. Hope that's ok

Reply:
Thanks @mhess-swl

Reply:
I expect these tests will continue to fail, because there is no setup for all the TSS messaging inside of `setCandidateRoster` in the failing tests

Reply:
The unit tests are fixed.